### PR TITLE
feat(breaking): Infer Prettier config and add prefer-optional-chaining (breaking)

### PR DESCRIPTION
**Pull request by zamotany**
### Summary

- Removes hard-coded Prettier config in favour of letting the Prettier find the config.
- Adds `@typescript-eslint/prefer-optional-chaining`

### Test plan

Updated the test files.

### CODE REVIEW

File: node.js [hunk 1]
```diff
@@ -76,6 +70,7 @@ module.exports = {
           ERROR,
           { argsIgnorePattern: '^_' },
         ],
+        '@typescript-eslint/prefer-optional-chain': ERROR,
```
Comment (thymikee) [inline node.js]: How about WARN for now?

Comment (zamotany) [reply node.js]: I can do WARN, but that doesn't change the fact that inferring Prettier config will be a breaking change itself.

Comment (thymikee) [reply node.js]: Fair call

File: test/.prettierrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "singleQuote": true,
+  :trailingComma:": "es5"
```
Comment (thymikee) [inline test/.prettierrc]: Is this intentionally broken?

Comment (zamotany) [reply test/.prettierrc]:
```suggestion
  "trailingComma": "es5"
```

File: test/index.js [hunk 3]
```diff
@@ -2,7 +2,7 @@
 import React from 'react';
 
 type Props = {
-  isTruthy: string,
+  isTruthy: string
```
Comment (thymikee) [inline test/index.js]:
```suggestion
  isTruthy: string,
```

**Comment by thymikee** [comment]
cc @matthargett @satya164

**Pull request merged**